pallets/pallet-lottery/src/lib.rs

```diff
@@ -0,0 +1,1077 @@
+
+    #[pallet::storage]
+    #[pallet::getter(fn unclaimed_winnings_by_account)]
+    pub(super) type UnclaimedWinningsByAccount<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, BalanceOf<T>, OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn total_unclaimed_winnings)]
+    pub(super) type TotalUnclaimedWinnings<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    #[derive(Clone, Encode, Decode, TypeInfo)]
+    pub(super) struct UnstakingCollator<AccountId, BlockNumber> {
+        pub account: AccountId,
+        pub since: BlockNumber,
+    }
+
+    #[pallet::storage]
+    pub(super) type UnstakingCollators<T: Config> =
+        StorageValue<_, Vec<UnstakingCollator<T::AccountId, T::BlockNumber>>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn remaining_unstaking_balance)]
+    pub(super) type RemainingUnstakingBalance<T: Config> =
+        StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    #[derive(Clone, Debug, Eq, PartialEq, Encode, Decode, TypeInfo)]
+    pub struct Request<AccountId, BlockNumber, Balance> {
+        pub user: AccountId,
+        pub block: BlockNumber,
+        pub balance: Balance,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn withdrawal_request_queue)]
+    pub(super) type WithdrawalRequestQueue<T: Config> =
+        StorageValue<_, Vec<Request<T::AccountId, T::BlockNumber, BalanceOf<T>>>, ValueQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn staked_collators)]
+    pub(super) type StakedCollators<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, BalanceOf<T>, ValueQuery>;
+
+    #[pallet::genesis_config]
+    pub struct GenesisConfig<T: Config> {
+        /// amount of token to keep in the pot for paying gas fees
+        pub gas_reserve: BalanceOf<T>,
+        pub min_deposit: BalanceOf<T>,
+        pub min_withdraw: BalanceOf<T>,
+    }
+
+    #[cfg(feature = "std")]
+    impl<T: Config> Default for GenesisConfig<T> {
+        fn default() -> Self {
+            Self {
+                min_deposit: 1u32.into(),
+                min_withdraw: 1u32.into(),
+                gas_reserve: 10_000u32.into(),
+            }
+        }
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+        #[inline]
+        fn build(&self) {
+            GasReserve::<T>::set(self.gas_reserve);
+            MinDeposit::<T>::set(self.min_deposit);
+            MinWithdraw::<T>::set(self.min_withdraw);
+        }
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(fn deposit_event)]
+    pub enum Event<T: Config> {
+        LotteryStarted,
+        LotteryStopped,
+        LotteryWinner(T::AccountId, BalanceOf<T>),
+        Deposited(T::AccountId, BalanceOf<T>),
+        ScheduledWithdraw(T::AccountId, BalanceOf<T>),
+        Withdrawn(T::AccountId, BalanceOf<T>),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        LotteryNotStarted,
```

Comment:
Same here, comments are missing. Once error happens, we can see some more details from blocks(polkadot-js)

Reply:
Thx Jamie, added the comments and removed a few unused error types